src/serialize/value.rs

```diff
@@ -0,0 +1,1050 @@
+            NativeType::Counter => {
+                let value = self
+                    .extract::<i64>()
+                    .map_err(|_| self.mismatched_type_error::<Counter>(typ))?;
+
+                let counter = Counter(value);
+                counter.serialize(typ, cell_writer)
+            }
+            NativeType::Varint => {
+                // Number of bits necessary to represent an integer in binary,
+                // EXCLUDING the sign and leading zeros.
+                let bits_len = self
+                    .call_method0("bit_length")
+                    .and_then(|x| x.extract::<usize>())
+                    .map_err(|_| self.mismatched_type_error::<CqlVarint>(typ))?;
+
+                // The `to_bytes()` method requires the number of bytes necessary to represent
+                // the given integer. The '+1' accounts for values such as 128, where `bits_len / 8 = 1`,
+                // but two's-complement representation in big-endian order (CqlVarintBorrowed format)
+                // requires 2 bytes, since a single signed byte covers only the range -128..127.
+                let bytes_len = bits_len / 8 + 1;
+
+                // TODO: I think there might be a way to create PyDict only once
+                // and reuse it in every `Varint` serialization.
+                let py_dict = PyDict::new(self.py());
+                py_dict.set_item("signed", true).map_err(|e| {
+                    SerializationError::new(PythonDriverSerializationError::PythonError(e))
+                })?;
+
+                // CqlVarintBorrowed is represented as two's-complement binary in big-endian order
+                // and Python's to_bytes() with byteorder='big' and signed=True also returns
+                // bytes in this format.
+                let bytes = self
+                    .call_method("to_bytes", (bytes_len, "big"), Some(&py_dict))
+                    .map_err(|_| self.mismatched_type_error::<CqlVarint>(typ))?;
+
+                let bytes = bytes
+                    .downcast::<PyBytes>()
+                    .map_err(|_| self.mismatched_type_error::<CqlVarint>(typ))?;
+
+                let bytes = bytes.as_bytes();
+                let varint = CqlVarintBorrowed::from_signed_bytes_be_slice(bytes);
+                varint.serialize(typ, cell_writer)
+            }
+
+            // Float types.
+            NativeType::Float => self.serialize_native::<f32>(typ, cell_writer),
+            NativeType::Double => self.serialize_native::<f64>(typ, cell_writer),
+            NativeType::Decimal => {
+                let value = self
+                    .extract::<BigDecimal>()
+                    .map_err(|_| self.mismatched_type_error::<CqlDecimal>(typ))?;
+
+                let decimal = CqlDecimal::try_from(value)
+                    .map_err(|_| SerializationError::new(ValueOverflow))?;
+                decimal.serialize(typ, cell_writer)
```

Comment:
:wrench: You should be able to directly serialize `BigDecimal`, as it's supported by the Rust Driver.